src/Client/Pages/Admin/UserForm.razor

```
@@ -56,19 +63,40 @@
         {
             Name = string.Empty,
             Password = string.Empty,
+            Email = string.Empty,
             Company = string.Empty,
             LatestPaymentDate = DateTime.UtcNow,
         };
     }
 
     private async Task ConvertAndUploadUser()
     {
+    //Make sure email is valid
+        if (!IsValidEmail(_user.Email))
+        {
+            return;
+        }
+
     // Make sure date is in UTC.
         _user.LatestPaymentDate = _user.LatestPaymentDate.ToUniversalTime();
 
         await UploadUser();
     }
 
+    private bool IsValidEmail(string email)
+    {
+        var trimmedEmail = email.Trim();
+        try
+        {
+            var addr = new MailAddress(email);
+            return addr.Address == trimmedEmail;
+        }
+        catch
```

It's bad practice to catch all exceptions, you should only catch the exceptions you specifically expect!

E.g.
```suggestion
        catch(FormatException)
```